### PR TITLE
Updates basic tutorials for Cadence 1.0 and fixes old migration commands

### PR DESCRIPTION
**Pull request by joshuahannan**
Closes https://github.com/onflow/docs/issues/769

I went through and made sure all the code looked good for the first 6 tutorials and that important new concepts like entitlements were explained properly.

I had to rush through the Fungible Token tutorial a bit since USDC stuff is the priority now, so it is a little sloppy, but at least it has updated code now.

The playground links are not updated yet because I'm waiting for either the Cadence 1.0 playground to become the flagship playground or when I finish updating the other tutorials, whichever comes first. Right now, the links still use old playground and old code, but the code examples in the tutorials themselves have Cadence 1.0 code to show how it works.

I also updated some of the commands in the migration guide to be correct

### CODE REVIEW

File: versioned_docs/version-1.0/cadence-migration-guide/index.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ Help is available during the [Cadence 1.0 Office Hours](https://calendar.google.
 #### Expectations
 
 - Any contracts that are not upgraded will fail after Testnet and Mainnet upgrade.
+  They **WILL NOT** be able to be upgraded again, so it is imperative
```
Comment (gregsantos) [inline versioned_docs/version-1.0/cadence-migration-guide/index.md:43]:
```suggestion
  They **cannot** be upgraded again, so it is imperative
```

File: versioned_docs/version-1.0/tutorial/01-first-steps.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Cadence introduces new features to smart contract programming that help develope
 - Resource-oriented programming, a new paradigm that pairs linear types with object capabilities to create a secure and declarative model for digital ownership
   by ensuring that resources (and their associated assets) can only exist in one location at a time, cannot be copied, and cannot be accidentally lost or deleted
 - Built-in pre-conditions and post-conditions for functions and [transactions](../language/transactions.md)
-- The utilization of capability-based security, which enforces access control by requiring that access to objects
+- The utilization of capability-based security and entitlements, which enforces access control by requiring that access to objects
```
Comment (gregsantos) [inline versioned_docs/version-1.0/tutorial/01-first-steps.md:18]:
```suggestion
- The utilization of capability-based security and entitlements, which enforces access control by requiring that access to objects,
```

File: versioned_docs/version-1.0/tutorial/01-first-steps.md [hunk 3]
```diff
@@ -24,14 +24,19 @@ Please see the [Cadence introduction](../index.md) for more information about th
...
-an in-browser editor and emulator to experiment with Flow.
+The [Flow Playground](https://play.flow.com) includes
+an in-browser editor and Flow emulator to experiment with Flow.
 Using the Flow Playground, you can write Cadence smart contracts,
 deploy them to a local Flow emulated blockchain, and submit transactions.
 
 The Flow Playground should be compatible with any standard web browser,
 but we recommend that you use Google Chrome with it,
-because it has been tested and optimized for only the Chrome browser so far.
+because it has been tested and optimized primarily with the Chrome browser so far.
```
Comment (gregsantos) [inline versioned_docs/version-1.0/tutorial/01-first-steps.md:34]: The Flow Playground should work with any standard web browser. However, we recommend using Google Chrome, as it has been primarily tested and optimized for this browser.

File: versioned_docs/version-1.0/tutorial/01-first-steps.md [hunk 4]
```diff
@@ -24,14 +24,19 @@ Please see the [Cadence introduction](../index.md) for more information about th
...
+
+The [Flow Crescendo (Cadence 1.0) upgrade](https://flow.com/upgrade/crescendo)
+introduces new powerful features to Cadence. Currently, the playground uses an older version
```
Comment (gregsantos) [inline versioned_docs/version-1.0/tutorial/01-first-steps.md:37]:
```suggestion
introduces new, powerful features to Cadence. Currently, the playground uses an older version
```

File: versioned_docs/version-1.0/tutorial/01-first-steps.md [hunk 5]
```diff
@@ -40,20 +45,70 @@ with deploying Cadence smart contracts and interacting with transaction and scri
...
+
+## What is a smart contract?
 
-When you click on one of these links, the tutorial will open in a new tab
-and the contracts, transactions, and scripts will be loaded into the templates in the Playground for you to use.
+In regular terms, a contract is an agreement between two parties for some exchange of information or assets.
+Normally, the terms of a contract are supervised and enforced by a trusted third party, such as a bank or a lawyer.
+
+A smart contract is a computer program stored in a network like a blockchain
+that verifies and executes the performance of a contract (like a lawyer does)
```
Comment (gregsantos) [inline versioned_docs/version-1.0/tutorial/01-first-steps.md:75]:
```suggestion
that verifies and executes the performance of a contract
```

File: versioned_docs/version-1.0/tutorial/01-first-steps.md [hunk 6]
```diff
@@ -40,20 +45,70 @@ with deploying Cadence smart contracts and interacting with transaction and scri
...
+without the need for any trusted third party anywhere in the process, because the code itself is trusted.
+
+Programs that run on blockchains are commonly referred to as smart contracts
+because they mediate important functionality (such as currency)
+without having to rely on a central authority (like a bank).
```
Comment (gregsantos) [inline versioned_docs/version-1.0/tutorial/01-first-steps.md:80]:
```suggestion
Programs that run on blockchains are commonly referred to as smart contracts because they facilitate important functions, such as managing digital currency, without relying on a central authority like a bank.
```

**Review by gregsantos — APPROVED** [review]
few copy nits but lg2G

File: versioned_docs/version-1.0/tutorial/03-resources.md [hunk 7]
```diff
@@ -529,26 +521,25 @@ Optionals allow developers to account for `nil` cases more gracefully.
 Here, we explicitly have to account for the possibility that the `helloResource` object we got with `load` is `nil`
 (because `load` will return `nil` if there is nothing there to load).
 
-Using `?` "unwraps" the optional, meaning that it gets the value if it is there, before calling `hello`,
-but only if the value isn't `nil`. If the value is nil, the `?` returns `nil`.
+We use the nil-coalescing operator (`??`) to "unwrap" the optional.
+This basically means that we are handling the case where the `load` method returns `nil`.
+If it returns `nil`. We `panic`, which will abort execution of the transaction
+with an error message. It is important for developers to always to provide detailed error messages
```
Comment (sisyphusSmiling) [inline versioned_docs/version-1.0/tutorial/03-resources.md:527]:
```suggestion
with an error message. It is important for developers to always provide detailed error messages
```

File: versioned_docs/version-1.0/tutorial/05-non-fungible-tokens-2.md [hunk 8]
```diff
@@ -371,32 +358,30 @@ Now, back to the tutorial!
...
+This is where an important layer of access control comes in.
 Cadence utilizes [capability security](../language/capabilities.md),
 which means that for any given object, a user is allowed to access a field or method of that object if they either:
 
 - Are the owner of the object
 - Have a valid reference to that field or method (note that references can only be created from capabilities, and capabilities can only be created by the owner of the object)
 
 When a user stores their NFT `Collection` in their account storage, it is by default not available for other users to access.
-A user's authorized account object (`AuthAccount`, which gives access to private storage)
+A user's authorized account object (`&Account`, which gives access to private storage)
```
Comment (sisyphusSmiling) [inline versioned_docs/version-1.0/tutorial/05-non-fungible-tokens-2.md:372]: > authorized account object

Maybe "entitled account"? Only suggesting because earlier it's mentioned that `&Account` is publicly accessible and grants access to /public/ namespace.

**Review by sisyphusSmiling — APPROVED** [review]
Looks great!

File: versioned_docs/version-1.0/tutorial/01-first-steps.md (hunk 2 above)
Comment (joshuahannan) [reply versioned_docs/version-1.0/tutorial/01-first-steps.md:18]: no comma needed here. We're saying that the access is restricted, not that capability security is restricted

File: versioned_docs/version-1.0/tutorial/01-first-steps.md (hunk 5 above)
Comment (joshuahannan) [reply versioned_docs/version-1.0/tutorial/01-first-steps.md:75]: I disagree with this one. I like the clarification.

File: versioned_docs/version-1.0/tutorial/05-non-fungible-tokens-2.md (hunk 8 above)
Comment (joshuahannan) [reply versioned_docs/version-1.0/tutorial/05-non-fungible-tokens-2.md:372]: That is a good point. I'll reword it to make it more clear

**Comment by joshuahannan** [comment]
I addressed all the comments. Doesn't seem like there is any blockers here, so I'm just going to merge. 

I created two more issues to track the rest of the work required for the tutorials and assigned them to myself, so I'll be able to get to those after the USDC work is done:
* https://github.com/onflow/docs/issues/800
* https://github.com/onflow/docs/issues/799
